### PR TITLE
api request limit fixes

### PR DESCRIPTION
**Pull request by amoffat**
makes our cooldown a little more effective.  also reenables caching of api requests that were chewing through our limit.

### CODE REVIEW

**Comment by PlasmaPower** [comment]
Was this causing the broken chaosbot behavior?

Edit: yep

**Comment by phil-r** [comment]
`./chaos.py:17:1: F401 'patch' imported but unused`

**Comment by PlasmaPower** [comment]
You can fix it with `# noqa: F401`

File: github_api/__init__.py [hunk 1]
```diff
@@ -14,7 +14,15 @@ def compute_api_cooldown(remaining, reset):
...
-    return ((reset / remaining) ** 3) / 10.0
+
+    # i've seen github's api fail with a 403 when we had 5 requests remaining,
+    # as confirmed by their response headers.  so we'll take that kind of
+    # fluctuation into account.
+    #
+    # also account for a potential divide by zero
+    actual_remaining = max(remaining - 30, 1)
+
+    return ((reset / actual_remaining) ** 3) / 10.0
```
Comment (PlasmaPower) [inline github_api/__init__.py]: Should we design this function to max out at `reset` when `actual_remaining == 1`?

Comment (amoffat) [reply github_api/__init__.py]: that's a great idea, let me do that

Comment (PlasmaPower) [reply github_api/__init__.py]: You might want to add a bit to reset just in case.

**Comment by PlasmaPower** [comment]
> make travis happy for now

Not quite - two spaces before an inline comment. Remember that you can run flake8 locally.

**Comment by amoffat** [comment]
Thanks guys :+1:

File: github_api/prs.py [hunk 2]
```diff
@@ -182,7 +182,7 @@ def get_ready_prs(api, urn, window):
         # existing on the PRs anymore (somehow..still unsolved), and then PRs
         # were not being processed or updated.  do not use this variable in the
         # if-condition that follow it until that has been solved
-        build_passed = has_build_passed(api, pr["statuses_url"])
+        build_passed = has_build_passed(api, pr["statuses_url"])  # noqa: F841
```
Comment (PlasmaPower) [inline github_api/prs.py]: Conflict here

**Pull request merged**